src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -1982,7 +1982,7 @@ private bool ScanIdentifier_CrefSlowPath(ref TokenInfo info)
                             else if (_identLen > 0 && consumedChar > 127 && SyntaxFacts.IsIdentifierPartCharacter(consumedChar))
                             {
                                 //// BUG 424819 : Handle identifier chars > 0xFFFF via surrogate pairs
-                                if (SyntaxFacts.IsFormattingChar(consumedChar))
+                                if (UnicodeCharacterUtilities.IsFormattingChar(consumedChar))
```

Comment:
Alternately, file an issue for that and I'll do it separately. We can consolidate lots of the VB implementation too once these helpers are shared.